pkg/infra/block_collector.go

```diff
@@ -11,39 +13,51 @@ import (
 // on a certain number of peers within network.
 type BlockCollector struct {
 	sync.Mutex
-	threshold int
-	total     int
-	registry  map[uint64]int
+	thresholdPeers int
+	totalPeers     int
+	totalTx        int
+	committedTx    int
+	registry       map[uint64]int
 }
 
 // NewBlockCollector creates a BlockCollector
-func NewBlockCollector(threshold int, total int) (*BlockCollector, error) {
+func NewBlockCollector(thresholdPeers int, totalPeers int, totalTx int) (*BlockCollector, error) {
```

Comment:
like we discussed, `BlockCollector` now is just another `worker` in tape model, and should consume a "job" channel, on which block flows in.